lib/caching-precompiler.js

```diff
@@ -0,0 +1,95 @@
+var cachingTransform = require('caching-transform');
+var fs = require('fs');
+var path = require('path');
+var md5Hex = require('md5-hex');
+var stripBom = require('strip-bom');
+
+module.exports = CachingPrecompiler;
+
+function CachingPrecompiler(cacheDir) {
+	if (!(this instanceof CachingPrecompiler)) {
+		throw new Error('CachingPrecompiler must be called with new');
```

Comment:
Yeah sure. Probably worth matching style. 

That said, after doing it the way you suggest for a long time, I am starting to think it's a mistake. ES2015 classes throw when function called, so magic auto-`new`ing is not future proof. Also, it introduces an easy to overlook failure point when refactoring the function signature. I know I have forgotten to update that statement inside the if condition before.